File: 백준/Gold/1484. 다이어트/다이어트.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int G = Integer.parseInt(st.nextToken());
		List<Long> ans = new ArrayList<>();

		long s = 1;
		long e = 1;
		while ( G >= e) {
			long now = e*e - s*s;
			if (now == G) {
				ans.add(e++);
				s++;
			} else if (now > G) {
				s++;
			} else {
				e++;
			}
		}
		
		if (ans.isEmpty()) {
			System.out.print(-1);
		} else {
			for (long a : ans) {
				System.out.println(a);
			}
		}
	}
}

File: 백준/Gold/1484. 다이어트/다이어트.py
G = int(input())
lst = []
s, e = 1, int(G**0.5)


while G > e:
    if e == s: # 같을때 위로 보냄
        e += 1
        continue
    if e**2 - s**2 == G:
        lst.append(e)
    if e**2 - s**2 > G:
        s += 1
    else:
        e += 1
if len(lst) == 0:
    print(-1)
else:
    for ans in lst:
        print(ans)
